client/src/components/search/Search.tsx

```diff
@@ -50,209 +13,182 @@ interface SearchResultItem {
   duration: string;
   path: string;
   coverUrl?: string;
+  album_id: string;
 }
 
 interface SearchResults {
   songs: SearchResultItem[];
-  artists: DBArtist[];
+  artists: [];
 }
 
 const SearchPage: React.FC = () => {
   const [results, setResults] = useState<SearchResults>({ songs: [], artists: [] });
-  const [searchParams, setSearchParams] = useSearchParams();
+  const [searchParams] = useSearchParams();
   const navigate = useNavigate();
   const [isMenuExpanded, setIsMenuExpanded] = useState(false);
   const [currentSong, setCurrentSong] = useState<SearchResultItem | null>(null);
   const [userId, setUserId] = useState<string>('');
+  const [errMsg, setErrMsg] = useState<string | null>(null);
+
+  const searchKeyword = searchParams.get('keyword') || '';
 
   useEffect(() => {
-    // Get userId from your auth system
     const fetchUserId = async () => {
-      try {
-        const userJson = localStorage.getItem('user');
-        if (userJson) {
-          const userData = JSON.parse(userJson);
-          setUserId(userData.user_id);
-        }
-      } catch (error) {
-        console.error('Error fetching user ID:', error);
+      const userJson = localStorage.getItem('user');
+      if (userJson) {
+        const userData = JSON.parse(userJson);
+        setUserId(userData.user_id);
       }
     };
     fetchUserId();
   }, []);
 
-  const searchKeyword = searchParams.get('keyword') || '';
-  const searchType = searchParams.get('type') || 'all';
-
   const handleSearch = async () => {
     if (searchKeyword.trim() === '') {
       setResults({ songs: [], artists: [] });
       return;
     }
 
+    const token = localStorage.getItem('token') || '';  // Retrieve token from local storage
+
     try {
-      // Make the search request to your API
-      const response = await fetch(`${import.meta.env.VITE_API_URL}/api/search`, {
-        method: 'POST',
+      const response = await axios.get(`http://localhost:8080/api/songs/search`, {
```

Comment:
When import Axios, and call axios is already populated. So when I deploy app this is going to cause an error. You should be able to use Axios.get('/songs/search' ....